rms_fe/src/views/ArticleDetailsPage/ArticleRecommendations.tsx

```diff
@@ -108,25 +110,101 @@ const AuthorPopover: React.FC<{ author: Author }> = ({ author }) => {
-    <>
-        <AuthorLink icon={<LinkOutlined />} url={author.scopus_url} label="Scopus Profile" />
-        <AuthorLink icon={<LinkOutlined />} url={"#"} label="DBLP Profile" />
-        <AuthorLink icon={<LinkOutlined />} url={"#"} label="Scolar Profile" />
-    </>
-);
 
-const AuthorLink: React.FC<{ icon: React.ReactNode; url?: string; label: string }> = ({ icon, url, label }) => {
-    if (!url) return null;
+const useDblpAuthor = (author_name: string) => {
+    return useQuery({ queryKey: ["author_dblp", author_name], queryFn: () => getAuthorDBLP(author_name), staleTime: 5 * 60 * 1000 })
+}
+
+const useScholarAuthor = (author_name: string) => {
+    return useQuery({ queryKey: ["author_scholar", author_name], queryFn: () => getAuthorScholar(author_name), staleTime: 5 * 60 * 1000 })
+}
+
+const AuthorLinks: React.FC<{ author: Author }> = ({ author }) => {
+
+    const authorName = author.given_name + " " + author.surname;
+    const dblpAuthor = useDblpAuthor(authorName);
+    const scholarAuthor = useScholarAuthor(authorName);
+
+    const handleScopusClick = async () => {
+        window.open(author.scopus_url, '_blank');
+    };
+
+    const handleDblpClick = async () => {
+        if (dblpAuthor.data?.authors.length === 1) {
+            window.open(dblpAuthor.data.authors[0].dblp_url, '_blank');
+        } else {
+            Modal.info({
+                title: 'We have found multiple DBLP profiles that match with the author name',
+                content: (
+                    <Flex vertical gap={5}>
+                        {dblpAuthor.data?.authors.map(author => (
+                            <AuthorLink key={author.dblp_url}
+                                icon={<LinkOutlined />}
+                                label={authorName} onClick={() => window.open(author.dblp_url, '_blank')} />
+                        ))}
+                    </Flex>
+                ),
+                onOk() { },
+            });
+        }
+    };
+
+    const handleScholarClick = async () => {
+        if (scholarAuthor.data?.authors.length === 1) {
+            window.open(scholarAuthor.data.authors[0].scholar_url, '_blank');
+        } else {
+            console.log(scholarAuthor.data?.authors)
+            console.log(scholarAuthor.data?.authors.length)
+            Modal.info({
+                title: 'We have found multiple Scholar profiles that match with the author name',
+                content: (
+                    <Flex vertical gap={5}>
+                        {scholarAuthor.data?.authors.map(author => (
+                            <AuthorLink key={author.scholar_url}
+                                icon={<LinkOutlined />}
+                                label={authorName} onClick={() => window.open(author.scholar_url, '_blank')} />
+                        ))}
+                    </Flex>
+                ),
+                onOk() { },
+            });
+        }
```

Comment:
This is practically exactly the same as it was for the dblp search, please extract it into a separate component